src/webui/www/private/scripts/client.js

```diff
@@ -426,35 +474,35 @@ window.addEventListener("DOMContentLoaded", (event) => {
     initializeWindows();
 
     // Show Top Toolbar is enabled by default
-    let showTopToolbar = localPreferences.get("show_top_toolbar", "true") === "true";
+    let showTopToolbar = clientData.get("show_top_toolbar") !== false;
```

Comment:
Continuing https://github.com/qbittorrent/qBittorrent/pull/23191#discussion_r2347245274

The code has become less straightforward.
Later someone will probably try to change it to (or write new code) `clientData.get("show_top_toolbar") === true` and only to find out it doesn't work.
This kind of usage hides a lot of elusive details and it just make the code harder to comprehend, maintain and write.

Reply:
This code is now _more_ straightforward. Rather than some additional parameter that applies in unknown situations, the caller can use native JS to get their default value.

See https://github.com/qbittorrent/qBittorrent/pull/23191#discussion_r2384388689

> `clientData.get("show_top_toolbar") === true` and only to find out it doesn't work reliably.

In what way does this not work reliably?

Reply:
I'm not sure I fully understand what you're getting at, but it seems more logical to me to use `value === true` than `value !== false` to determine if a value is true.

Reply:
> Looking at this code again, it still makes sense to me.

Yes, because you know its whole background in advance.

> Is this really that confusing?

At least it is questionable, as it does not follow the usual pattern `value === true`, and it requires additional effort to figure out what it does.


> `value !== false` is a way of saying "this statement should evaluate to true unless this value is explicitly set to false".

The "get_or" ("value_or") semantics makes the code at the calling place more obvious about the intention that the requested value may be missing, and therefore we use the default value.

Reply:
> > Is this really that confusing?
> 
> At least it is questionable, as it does not follow the usual pattern `value === true`, and it requires additional effort to figure out what it does.

And yes, there is a good chance (as my experience shows) that in the future someone will want to revamp this part of the code and break its original behavior if they rely only on "common sense" and do not delve deeply into its ins and outs.

Reply:
> I'm going to switch to the more verbose but more idiomatic way of doing this in JS - basically the JS equivalent of `value_or`. `clientData.get("show_search_engine") !== false` is now `(clientData.get("show_search_engine") ?? true) === true`.

👍